log-warper/src/Log/Event.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Contains implementations of logging records and events which are consumed by 'LogAction'.
+
+module Log.Event
+       ( -- * Types
+         Event (..)
+       , eName
+       , eSeverity
+       , eMessage
+       , eConfiguration
+
+       , RichEvent (..)
+       , reEvent
+       , reThread
+       , reTime
+
+         -- * Functions
+       , makeRich
+       ) where
+
+import Universum
+
+import Control.Concurrent (ThreadId, myThreadId)
+import Data.Time.Clock (UTCTime, getCurrentTime)
+import Lens.Micro.Platform (makeLenses)
+
+import Log.Configuration (Configuration, Extension, cGlobal, gpShowTid, gpShowTime)
+import Log.Core.Action (LogAction (..), cbind)
+import Log.Name (LoggerName (..))
+import Log.Severity (Severity (..))
+
+
+{- | Bundle of all basic logging data. This logging record also contains
+configuration for logging. It can be used to process messages. All 'LogAction's
+in @log-warper@ should have form:
+
+@
+'LogAction' someM ('Event' exts)
+@
+-}
+data Event (exts :: [Extension]) = Event
+    { _eName          :: !LoggerName
+    , _eSeverity      :: !Severity
+    , _eMessage       :: !Text
+    , _eConfiguration :: !(Configuration exts)
```

Comment:
I do not quite understand why is this field needed.

Reply:
@volhovm I explained why I think it's needed at the first half of this comment:

* https://github.com/serokell/log-warper/pull/108#discussion_r183084927